tablo/wkt.py

```diff
@@ -0,0 +1,185 @@
+# All the functions here are translations of ESRI Terraformer library from JS and are used for converting
+# ESRI geometry JSON to GeoJSON
+# TODO look for optimization possibilities
+
+
+def from_rings(rings):
+    """
+    Converts an ESRI polygon to GeoJSON.
+    :param rings: a collection of rings from an ESRI geometry JSON similar to the following:
+        .. code-block:: json
+
+            {
+                geometry: {
+                    rings: [
+                        [[5, 2], [6, 5], [2, 9], [-2, 5], [1, 2], [2, 3], [5, 2]],
+                        [[0, 0], [-5, 5], [0, 10], [5, 10], [10, 5], [5, 0], [0, 0]],
+                        [[1, 7], [3, 7], [3, 4], [1, 4], [1, 7]]
+                    ],
+                    'spatialReference': {'wkid': 4326}
+                }
+            }
+        see here for more info: http://help.arcgis.com/en/arcgisserver/10.0/apis/rest/geometry.html
+    :return: A GeoJSON object
+    """
+    outer_rings = []
+    holes = []
+
+    for r in rings:
+        ring = close_ring(r)
+        if len(ring) < 4:
```

Comment:
I still don't follow, wouldn't a ring with 3 unique coords (since first and last are dup) be a triangle?